lib/src/Uncertainty/Algorithm/Optimization/EfficientGlobalOptimization.cxx

```diff
@@ -302,14 +302,39 @@ void EfficientGlobalOptimization::run()
     }
     solver_.run();
     const OptimizationResult improvementResult(solver_.getResult());
-
     // store improvement
-    Point improvementValue(improvementResult.getOptimalValue());
-    expectedImprovement_.add(improvementValue);
+    Point newPoint(improvementResult.getOptimalPoint());
+    Point newValue;
+    Point improvementValue;
+    if (!problem.hasBounds() || problem.getBounds().contains(newPoint))
+      {
+        improvementValue = improvementResult.getOptimalValue();
+        expectedImprovement_.add(improvementValue);
+
```

Comment:
```suggestion
```

Reply:
The commit has been removed from the PR, the issue being solved in another PR.